app/controllers/errors_controller.rb

```diff
@@ -0,0 +1,10 @@
+class ErrorsController < ApplicationController
+  def file_not_found
```

Comment:
I think we can just have these controllers return `head :$status`. This gets rid of the exception type behavior that Rails is using to treat 404s, and keeps the return codes consistent